internal/strict/strict.go

```diff
@@ -58,78 +63,99 @@ func GetStrictControl(name string) (Control, bool) {
+		Name:    SkipDependenciesInputs,
+		Error:   errors.Errorf("The `%s` option is deprecated. Reading inputs from dependencies has been deprecated and will be removed in a future version of Terragrunt. To continue using inputs from dependencies, forward them as outputs.", SkipDependenciesInputs),
 		Warning: fmt.Sprintf("The `%s` option is deprecated and will be removed in a future version of Terragrunt. Reading inputs from dependencies has been deprecated. To continue using inputs from dependencies, forward them as outputs.", SkipDependenciesInputs),
 	},
 	DisableLogFormatting: {
-		Error:   errors.Errorf("The `--%s` flag is no longer supported. Use `--terragrunt-log-format=key-value` instead.", DisableLogFormatting), //nolint:revive
+		Name:    DisableLogFormatting,
+		Error:   errors.Errorf("The `--%s` flag is no longer supported. Use `--terragrunt-log-format=key-value` instead.", DisableLogFormatting),
 		Warning: fmt.Sprintf("The `--%s` flag is deprecated and will be removed in a future version. Use `--terragrunt-log-format=key-value` instead.", DisableLogFormatting),
 	},
 	JSONLog: {
-		Error:   errors.Errorf("The `--%s` flag is no longer supported. Use `--terragrunt-log-format=json` instead.", JSONLog), //nolint:revive
+		Name:    JSONLog,
+		Error:   errors.Errorf("The `--%s` flag is no longer supported. Use `--terragrunt-log-format=json` instead.", JSONLog),
 		Warning: fmt.Sprintf("The `--%s` flag is deprecated and will be removed in a future version. Use `--terragrunt-log-format=json` instead.", JSONLog),
 	},
 	TfLogJSON: {
-		Error:   errors.Errorf("The `--%s` flag is no longer supported. Use `--terragrunt-log-format=json` instead.", TfLogJSON), //nolint:revive
+		Name:    TfLogJSON,
+		Error:   errors.Errorf("The `--%s` flag is no longer supported. Use `--terragrunt-log-format=json` instead.", TfLogJSON),
 		Warning: fmt.Sprintf("The `--%s` flag is deprecated and will be removed in a future version. Use `--terragrunt-log-format=json` instead.", TfLogJSON),
 	},
+	RootTerragruntHCL: {
+		Name:    RootTerragruntHCL,
+		Error:   errors.Errorf("Using `terragrunt.hcl` as the root of Terragrunt configurations is an anti-pattern, and no longer supported. Use a differently named file like `root.hcl` instead. For more information, see https://terragrunt.gruntwork.io/docs/migrate/migrating-from-root-terragrunt-hcl"),
+		Warning: "Using `terragrunt.hcl` as the root of Terragrunt configurations is an anti-pattern, and no longer recommended. In a future version of Terragrunt, this will result in an error. You are advised to use a differently named file like `root.hcl` instead. For more information, see https://terragrunt.gruntwork.io/docs/migrate/migrating-from-root-terragrunt-hcl",
+	},
 }
 
+var TriggeredControls = xsync.NewMapOf[string, bool]()
```

Comment:
Global variable can cause problems in tests, especially with `-race`.

Reply:
Ya, but does it really matter? 

As long as we test the stderr check integration test out of parallel, we should be fine, right?
https://github.com/gruntwork-io/terragrunt/pull/3588/files#diff-4e1bb7ad3f548fa9a54e7e0f4b57c046d000c335ba4802df083a8e93416d5abfR85

In unit tests, I just ignore the `triggered` status:
https://github.com/gruntwork-io/terragrunt/pull/3588/files#diff-5e4aa2fd41e755849a1962333f14613e6189d0aaf00d10eac0a756cebe1692ccR68

Reply:
Until we create another test to check the strict-mode (in config or CLI), this should work stably. It's up to you, we just need remember this.